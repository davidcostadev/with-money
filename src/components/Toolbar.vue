<template>
  <md-toolbar>
    <md-button id="btn-previus-month" @click="toPreviousMonth">
      <md-icon>chevron_left</md-icon> {{previous_month_str}}
    </md-button>
    <h3 class="md-title" style="flex: 1; text-align: center;">{{state_month_str}}</h3>
    <md-button id="btn-next-month" @click="toNextMonth">
      <md-icon>chevron_right</md-icon> {{next_month_str}}
    </md-button>
  </md-toolbar>
</template>

<script>
import { mapGetters } from 'vuex';

export default {
  computed: {
    ...mapGetters([
      'state_month_str',
      'next_month_str',
      'previous_month_str',
    ]),
  },
  methods: {
    toPreviousMonth() {
      this.$store.dispatch('toPreviousMonth');
      this.$emit('update:stateMonth');
    },
    toNextMonth() {
      this.$store.dispatch('toNextMonth');
      this.$emit('update:stateMonth');
    },
  },
};
</script>
