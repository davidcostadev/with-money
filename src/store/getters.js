export default {
  user(state) {
    return state.user;
  },
  current_month(state) {
    return state.current_month;
  },
  state_month(state) {
    return state.state_month;
  },
  state_month_str(state) {
    return state.state_month_str;
  },
  next_month_str(state) {
    return state.next_month_str;
  },
  previous_month_str(state) {
    return state.previous_month_str;
  },
};
