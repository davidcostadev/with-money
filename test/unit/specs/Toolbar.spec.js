import Moment from 'moment';
import VueMaterial from 'vue-material';
import { createLocalVue, shallowMount } from '@vue/test-utils';
import Vuex from 'vuex';
import Toolbar from '../../../src/components/Toolbar';
import getters from '../../../src/store/getters';

const localVue = createLocalVue();


localVue.use(Vuex);
localVue.use(VueMaterial);

describe('Toolbar Component', () => {
  let store;
  let now;

  beforeEach(() => {
    store = new Vuex.Store({
      state: {
        current_month: Moment(),
        state_month: Moment(),
        state_month_str: Moment().format('MMM YY'),
        next_month_str: Moment().add(1, 'month').format('MMM YY'),
        previous_month_str: Moment().subtract(1, 'month').format('MMM YY'),
      },
      getters,
    });
  });

  beforeAll(() => {
    now = Date.now;
    Date.now = () => 1532473578215;
  });
  afterAll(() => {
    Date.now = now;
  });

  describe('Render', () => {
    it('default', () => {
      const wrapper = shallowMount(Toolbar, { store, localVue });

      expect(wrapper.isVueInstance).toBeTruthy();
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
