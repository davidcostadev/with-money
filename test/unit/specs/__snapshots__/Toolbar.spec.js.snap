// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toolbar Component Render default 1`] = `
<mdtoolbar-stub>
  <mdbutton-stub
    id="btn-previus-month"
  >
    <mdicon-stub>
      chevron_left
    </mdicon-stub>
     June 18
  
  </mdbutton-stub>
   
  <h3
    class="md-title"
    style="text-align: center;"
  >
    July 18
  </h3>
   
  <mdbutton-stub
    id="btn-next-month"
  >
    <mdicon-stub>
      chevron_right
    </mdicon-stub>
     August 18
  
  </mdbutton-stub>
</mdtoolbar-stub>
`;
